util/argo/argo.go

```diff
@@ -455,3 +458,54 @@ func getDestinationServer(ctx context.Context, db db.ArgoDB, clusterName string)
 	}
 	return servers[0], nil
 }
+
+func GetAppVirtualProject(proj *argoappv1.AppProject, settingsManager *settings.SettingsManager, appclientset appclientset.Interface) (*argoappv1.AppProject, error) {
```

Comment:
We need to merge global project settings pretty much everywhere in appcontroller, not only in sync.go. Likely there is already `GetAppProject ` method that does it: 

https://github.com/argoproj/argo-cd/blob/701dda9a2877742a6ef17410de08bd5008368284/util/argo/argo.go#L320-L322


Instead of adding `GetAppVirtualProject` and `mergeVirtualProject` methods I suggest to implement merging right in `GetAppProject` method. ( we would have to replace `ns` parameter with `settingsManager *settings.SettingsManager` )